src/components/ToDoList.tsx

```diff
@@ -20,29 +20,39 @@ export const ToDoList = React.memo((props: ListsProps) => {
     null
   );
 
-  const addItem = () => {
+  const addItem = React.useCallback(() => {
     if (newText.trim()) {
       setItems([...items, { id: Date.now(), text: newText }]);
       setNewText("");
     }
-  };
-  const updateItem = () => {
+  }, [newText, setItems, setNewText]);
+
+  const updateItem = React.useCallback(() => {
     setItems(
       items.map((t) => (t.id === newTextWithId?.id ? newTextWithId : t))
     );
     setNewTextWithId(null);
-  };
-  const deleteItem = (id: number) => {
-    setItems(items.filter((item) => item.id !== id));
-  };
+  }, [setItems, newTextWithId, setNewTextWithId]);
+
+  const deleteItem = React.useCallback(
+    (id: number) => {
+      setItems(items.filter((item) => item.id !== id));
+    },
+    [items]
+  );
+
+  const onInputChange = React.useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => setNewText(e.target.value),
```

Comment:
`setNewText(e.target.value) -> setNewText(e.target.value ?? '')`
You could also double check the `toString`, `trim` here, validate etc.

In real life, people usually use libraries, for example:
https://react-hook-form.com/docs/useform
https://www.npmjs.com/package/yup